packages/ui/components/form/select/Select.tsx

```diff
@@ -15,6 +17,28 @@ export type SelectProps<
   Group extends GroupBase<Option> = GroupBase<Option>
 > = Props<Option, IsMulti, Group> & { variant?: "default" | "checkbox"; "data-testid"?: string };
 
+const Option = <
+  Option,
+  IsMulti extends boolean = false,
+  Group extends GroupBase<Option> = GroupBase<Option>
+>({
+  data,
+  ...props
+}: OptionProps<Option, IsMulti, Group>) => {
+  return (
+    <components.Option data={data} {...props}>
+      <>
+        {props.children}
+        {(data as { accepted?: boolean })?.accepted === false && (
```

Comment:
Kinda hacky, was getting type errors, couldn't figure out how to resolve those.

Reply:
I think changing the common UI component is not a good idea.
we can instead pass the component prop to the <Select/>
take a look how option is implementd

https://github.com/SomayChauhan/cal.com/blob/f3cb6d28a91244c1416b56667517aa6d4e012b8d/apps/web/components/eventtype/EventAvailabilityTab.tsx#L31

Reply:
That is what I was trying to find. Made changes tests should pass now.